create_blog_post_using_azure_openai_and_publish_in_medium/main.bal

```diff
@@ -0,0 +1,49 @@
+import ballerina/http;
+import ballerina/io;
+import ballerinax/azure_openai.text;
+import ballerinax/medium;
+
+configurable string openAIToken = ?;
+configurable string serviceUrl = ?;
+configurable string deploymentId = ?;
+configurable string mediumToken = ?;
+
+const API_VERSION = "2023-03-15-preview";
+
+public function main(string title) returns error? {
+    final medium:Client medium = check new ({auth: {token: mediumToken}});
+    final text:Client azureOpenAI = check new (
+        config = {httpVersion: http:HTTP_1_1, auth: {apiKey: openAIToken}},
+        serviceUrl = serviceUrl
+    );
+
+    medium:UserResponse response = check medium->getUserDetail();
+    medium:User? user = response.data;
+
+    if user !is medium:User {
+        return error("User is not a medium user");
+    }
+    string? userId = user.id;
+    if userId !is string {
+        return error("Medium user ID is not valid");
```

Comment:
Can't we do both checks at once?

Reply:
We can't directly access the userId as `string userId = response.data.id;` because response.data can be `None` and it doesn't support field access using the dot operator. Any other suggestions?